src/main/resources/Diagram/DiagramViewSheet.xml

```diff
@@ -301,7 +310,10 @@ require.config({
       lightbox: false
     }, options);
     return this.each(function() {
-      var xmlDoc = mxUtils.parseXml($(this).data('model'));
+      var xmlDoc = getDiagramFromURL();
+      if (null === xmlDoc) {
+        var xmlDoc = mxUtils.parseXml($(this).data('model'));
```

Comment:
In both cases you need to parse the XML in order to obtain an ``xmlDoc`` so I would have created a function ``getDiagramXML`` that tries to get the XML from URL and falls back on the XML read from the ``data-model`` attribute.